tasks/add-unique-granuleID/schemas/input.json

```diff
@@ -0,0 +1,27 @@
+{
+  "title": "AddUniqueGranuleIDInput",
+  "description": "Describes the input expected by the add-unique-granule-id task component",
+  "type": "object",
+  "required": ["granules"],
+  "properties": {
+    "granules": {
+      "type": "array",
+      "description": "Array of all granule objects with a granuleId. This should be an API formatted granule",
+      "items": {
+        "type": "object",
+        "required": ["granuleId"],
+        "properties": {
+          "granuleId": { "type": "string" },
+          "collectionId": { "type": "string" },
+          "producerGranuleId": { "type": "string" },
+          "dataType": { "type": "string" },
+          "version": { "type": "string" }
+        },
+        "anyOf": [
```

Comment:
Potentially dumb question. Is this saying we're expecting either an API-formatted granule with collectionId or a ...payload?... granule that has dataType + version? Why might we have one vs. the other?

Reply:
The difference is if this lambda is utilized before or after sync-granules. 

I'd expect it's generally used with a dataType/version granule, but we don't know how users are using the system - it's possible they're running ingest entirely without sync granules with pre-staged data, for example.